### PR TITLE
Disable and enable monitoring

### PR DESCRIPTION
**Pull request by leosoto**

### CODE REVIEW

**Comment by leosoto** [comment]
@luelher @afiebig @eugenioLopezRamos Do you guys agree with codeclimate here? https://codeclimate.com/github/e-gob/interoperabilidad/pull/39. It says the cognitive load is high for the method and I don't think so; but maybe that's because I wrote it. Would you suggest refactoring it? (and how?) or should I override CC's judgement?

File: app/controllers/monitoring/services_controller.rb [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+    @organization ||= Organization.where(name: params[:organization_name]).first
+  end
+
+  def service
+    @service ||= organization.services.where(name: params[:name]).first
+  end
+
+  def check_service_admin
+    unless current_user&.is_service_admin?
+      redirect_to monitoring_organization_services_path(organization)
```
Comment (afiebig) [inline app/controllers/monitoring/services_controller.rb]: shouldn't we flash a message when redirecting? `notice: t(:not_enough_permissions)` or something

**Comment by luelher** [comment]
@leosoto I Don't see the "complexity" of the code, I think that CC is over reacting, but I don't know if we can make CC more flexible in this matter instead of disable it. cc @afiebig @eugenioLopezRamos

File: app/views/monitoring/services/index.html.haml [hunk 2]
```diff
@@ -15,18 +15,37 @@
...
+                          = t(:monitoring_disabled)
+                    - if current_user&.is_service_admin?
+                      %td
+                        - if service.monitoring_enabled
+                          = link_to 'Desactivar Monitoreo', |
+                            disable_monitoring_organization_service_path(@organization, service), |
+                            data: {method: 'post'}, |
+                            class: 'btn btn-default'
+                        - else
+                          = link_to 'Activar Monitoreo', |
```
Comment (afiebig) [inline app/views/monitoring/services/index.html.haml]: We should Translate this using I18n

File: app/views/monitoring/services/index.html.haml [hunk 3]
```diff
@@ -15,18 +15,37 @@
...
-                        =t(service.current_or_last_version.availability_status)
+                      %span{class: css_class_for_availability_status(service)}
+                        - if service.monitoring_enabled
+                          = t(service.current_or_last_version.availability_status)
+                        - else
+                          = t(:monitoring_disabled)
+                    - if current_user&.is_service_admin?
+                      %td
+                        - if service.monitoring_enabled
+                          = link_to 'Desactivar Monitoreo', |
```
Comment (afiebig) [inline app/views/monitoring/services/index.html.haml]: We should Translate this using I18n

**Comment by eugenioLopezRamos** [comment]
The complexity thing seems excessive to me

File: app/controllers/monitoring/services_controller.rb (hunk 1 above)
Comment (leosoto) [reply app/controllers/monitoring/services_controller.rb]: 👍

**Pull request merged**